framework/rpc/grpc.py

```diff
@@ -95,10 +95,15 @@ def __init__(self, rpc_host):
         # Cache gRPC channels per port
         self.channels = dict()
 
-    def _make_channel(self, port) -> grpc.Channel:
+    def _make_channel(self, port, secure_mode=False) -> grpc.Channel:
```

Comment:
I really appreciate this change you made! This looks great!

One thing I'm afraid of is that `secure_mode` might get conflated with security test's server's "secure_mode", where we listen on different ports - one with tls, another plain-text.

May I ask you to rename `secure_mode` to `secure_channel` here and elsewhere? Just the parameter, services like `secure_csds` are good.